common:
  img_topic: "/flir_driver/camera1/image"
  lid_topic: "/points"
  imu_topic: "/imu"
  img_en: 1
  lidar_en: 1
  ros_driver_bug_fix: false

extrin_calib:
  extrinsic_T: [-0.006253, 0.011775, 0.03055]
  extrinsic_R: [-1, 0, 0, 0, -1, 0, 0, 0, 1]
  # NTU_VIRAL
  Rcl: [0.00015717779, -0.01599804877, 0.99987201067,
     -0.99999357035, -0.00358459191, 0.00009984308,
      0.00358253582, -0.99986559754, -0.01599850933]
  Pcl: [0.08372735744, 0.02091938060, -0.13379192488]

time_offset: #일단 대충
  lidar_time_offset: -0.1
  imu_time_offset: 0.0
  img_time_offset: 0.0
  exposure_time_init: 0.0

preprocess: 
  point_filter_num: 3
  filter_size_surf: 0.1
  lidar_type: 3 # Ouster
  scan_line: 128
  blind: 1.0

vio: #일단 대충
  max_iterations: 5
  outlier_threshold: 1000 # 78 100 156 #100 200 500 700 infinite
  img_point_cov: 100 # 100 1000 
  patch_size: 8
  patch_pyrimid_level: 3
  normal_en: true
  raycast_en: false
  inverse_composition_en: false
  exposure_estimate_en: true
  inv_expo_cov: 0.1

imu: #일단 ig_lio/config/ncd.yaml 에서 가져옴

  imu_en: true
  imu_int_frame: 30
  acc_cov: 0.001249
  gyr_cov: 0.000208
  b_acc_cov: 0.000106
  b_gyr_cov: 0.000004

lio:
  max_iterations: 5
  dept_err: 0.02
  beam_err: 0.01
  min_eigen_value: 0.0025 # 0.0025
  voxel_size: 0.5
  max_layer: 2
  max_points_num: 50
  layer_init_num: [5, 5, 5, 5, 5]

local_map:  
  map_sliding_en: false
  half_map_size: 100
  sliding_thresh: 8

uav:
  imu_rate_odom: false
  gravity_align_en: false

publish:
  dense_map_en: true
  pub_effect_point_en: false 
  pub_plane_en: false
  pub_scan_num: 1
  blind_rgb_points: 0.0

evo:
  seq_name: "eee_01"
  pose_output_en: true

pcd_save:
  pcd_save_en: false
  colmap_output_en: false # need to set interval = -1
  filter_size_pcd: 0.15
  interval: -1
  # how many LiDAR frames saved in each pcd file;
  # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
